data/src/main/java/com/chan/moviesearcher/data/DateJsonAdapter.kt

```diff
@@ -0,0 +1,38 @@
+package com.chan.moviesearcher.data
+
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.JsonReader
+import com.squareup.moshi.JsonWriter
+import java.io.IOException
+import java.text.SimpleDateFormat
+import java.util.*
+
+class DateJsonAdapter : JsonAdapter<Date>() {
+
+    private val dateFormat = SimpleDateFormat(NAVER_DATE_FORMAT, Locale.US)
+
+    @Synchronized
+    @Throws(IOException::class)
+    override fun fromJson(reader: JsonReader): Date {
+        return if (reader.peek() == JsonReader.Token.NULL) {
+            reader.nextNull<Any>() as Date
+        } else {
+            val string = reader.nextString()
+            return dateFormat.parse(string)
+        }
+    }
+
+    @Synchronized
+    @Throws(IOException::class)
+    override fun toJson(writer: JsonWriter, value: Date?) {
```

Comment:
P3. 여기도 nullable 제거해주세요
```suggestion
    override fun toJson(writer: JsonWriter, value: Date) {
```

Reply:
상위 클래스가 아래와 같아서 nullable을 제거 할 수 없습니다.
```
public abstract void toJson(JsonWriter writer, @Nullable T value) throws IOException;
```